contracts/StakeManager.sol

```diff
@@ -127,31 +165,33 @@ contract StakeManager is Ownable {
      * @dev Reverts when `_time` is bigger than `MAX_LOCKUP_PERIOD`
      * @dev Reverts when `_time + block.timestamp` is smaller than current lock time.
      */
-    function lock(uint256 _time) external onlyVault noMigration {
+    function lock(uint256 _time) external onlyVault noMigration processEpoch {
         if (_time > MAX_LOCKUP_PERIOD) {
             revert StakeManager__InvalidLockupPeriod();
         }
         Account storage account = accounts[msg.sender];
         processAccount(account, currentEpoch);
-        if (block.timestamp + _time < account.lockUntil) {
-            revert StakeManager__DecreasingLockTime();
-        }
-        mintIntialMultiplier(account, _time, account.balance, 0);
+        mintIntialMP(account, _time, 0);
+        //update account storage
+        account.lockUntil += _time;
     }
-
     /**
      * @notice Release rewards for current epoch and increase epoch.
      */
-    function executeEpoch() external noMigration {
-        processEpoch();
+
+    function executeEpoch() external noMigration processEpoch {
+        return;
```

Comment:
Cool if this works, but I wonder if we should rather have an internal `_processEpoch()` function, which is used in the `processEpoch modifier`. This allows us to add `_processEpoch()` as function call here as well.

The reason I'm proposing this (and feel free to disagree), is that, at a first glance, this function here simply returns and it might be confusing. One has to know that `processEpoch modifier` is doing the actual work here.

Reply:
A "prerequisite" is a modifier that sets up the environment for a function to execute, rather than checking if a certain condition is true. 

For example, a Solidity developer might use a function that requires a certain amount of Ether to be sent along with it in order to execute. In that case, the prerequisite would be the function that sets up the Ether balance.

Reply:
I think is good, because actually it does nothing, just bumps the epoch, which would automatically happen in many other calls that have processEpoch modifier. 
We could even remove this external function, as I don't see anyone calling it instead of calling executeAccount for example.